### PR TITLE
Feature/fu 385

### PR DESCRIPTION
**Pull request by amitmo50**

### CODE REVIEW

File: projects/fusion-ui/components/chip-filter/common/base/chip-filter.base.component.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        this.onRemove.emit();
+        if (this.suppressClickOnRemove) {
+            $event.stopPropagation();
+        }
+    }
+
+    setClickListener(): void {
+        this.onDestroy$.next();
+        fromEvent(this.element.nativeElement, 'click')
+            .pipe(takeUntil(this.onDestroy$))
```
Comment (danielbrazg) [inline projects/fusion-ui/components/chip-filter/common/base/chip-filter.base.component.ts:88]: you should use a new subject for that managing the clickListener reset. and then use both OnDestroy & the new subject to reset the event listener:
for example:
```
this.resetListeners$.next();
fromEvent(this.element.nativeElement, 'click')
 .pipe(takeUntil(this.onDestroy$),takeUntil(this.resetListeners$))...
```

File: projects/fusion-ui/components/chip-filter/common/base/chip-filter.base.component.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    closeClicked($event) {
+        this.renderer.removeChild(this.renderer.parentNode(this.element.nativeElement), this.element.nativeElement);
+        this.onRemove.emit();
+        if (this.suppressClickOnRemove) {
+            $event.stopPropagation();
+        }
+    }
+
+    setClickListener(): void {
+        this.onDestroy$.next();
```
Comment (danielbrazg) [inline projects/fusion-ui/components/chip-filter/common/base/chip-filter.base.component.ts:86]: why do you need this?

File: projects/fusion-ui/components/chip-filter/v3/chip-filter.component.scss [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    width: fit-content;
+    height: $chipHeight;
+    @extend %chip-font;
+    border-radius: 100px;
+    border: none;
+    gap: 4px;
+    background-color: $fu-primary-100;
+    color:$fu-primary-500;
+
+    @include icon-style('fu-icon-close', $fu-primary-300, 16px, true, true,true){
```
Comment (danielbrazg) [inline projects/fusion-ui/components/chip-filter/v3/chip-filter.component.scss:37]: didn't we change the way we work with the icon component? why should still use ng-deep when we use icons?

File: projects/fusion-ui/components/chip-filter/v3/chip-filter.component.scss [hunk 4]
```diff
@@ -0,0 +1,125 @@
+@import '../../../src/style/scss/v3/mixins/mixins';
+@import '../../../src/style/scss/v3/vars/colors';
+@import '../../../src/style/scss/v3/vars/fonts';
+
+
+%chip-font {
```
Comment (danielbrazg) [inline projects/fusion-ui/components/chip-filter/v3/chip-filter.component.scss:6]: this is not good. fonts shouldn't define in components. it should be a library setting. please use one of our fonts. if it not included then let design team know

File: projects/fusion-ui/components/chip-filter/v3/chip-filter.component.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import {ChangeDetectionStrategy, Component, OnDestroy, OnInit} from '@angular/core';
+import {ChipFilterBaseComponent} from '@ironsource/fusion-ui/components/chip-filter/common/base';
+
+@Component({
+    selector: 'fusion-chip-filter',
+    templateUrl: './chip-filter.component.html',
+    styleUrls: ['./chip-filter.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ChipFilterComponent extends ChipFilterBaseComponent implements OnInit, OnDestroy {}
```
Comment (danielbrazg) [inline projects/fusion-ui/components/chip-filter/v3/chip-filter.component.ts:10]: you can remove `implements OnInit, OnDestroy` when you extend class that implement those classes

File: projects/fusion-ui/components/chip-filter/common/base/chip-filter.base.component.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    @Input() set disabled(disabled: boolean) {
+        this._disabled = disabled;
+        this.changeHostClass('fu-disabled', disabled);
+    }
+
+    get disabled(): boolean {
+        return this._disabled;
+    }
+
+    @Input()
```
Comment (danielbrazg) [inline projects/fusion-ui/components/chip-filter/common/base/chip-filter.base.component.ts:45]: why do you have also @input for specific configuration and also for @input configuration

**Review by danielbrazg — CHANGES_REQUESTED** [review]

File: projects/fusion-ui/components/chip-filter/v3/chip-filter.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import {waitForAsync, ComponentFixture, TestBed} from '@angular/core/testing';
+import {ChipFilterComponent} from '@ironsource/fusion-ui/components/chip-filter';
+import {TooltipModule, IconModule, FlagModule} from '@ironsource/fusion-ui';
+
+describe('ChipFilterComponent', () => {
+    let component: ChipFilterComponent;
+    let fixture: ComponentFixture<ChipFilterComponent>;
```
Comment (danielbrazg) [inline projects/fusion-ui/components/chip-filter/v3/chip-filter.component.spec.ts:7]: add basic unit tests

**Review by danielbrazg — CHANGES_REQUESTED** [review]

File: projects/fusion-ui/components/chip-filter/common/base/chip-filter.base.component.ts (hunk 2 above)
Comment (amitmo50) [reply projects/fusion-ui/components/chip-filter/common/base/chip-filter.base.component.ts:86]: my mistake

**Pull request merged**